libraries/AP_Compass/AP_Compass.cpp

```diff
@@ -1175,8 +1211,10 @@ bool Compass::configured(uint8_t i)
 bool Compass::configured(void)
 {
     bool all_configured = true;
-    for(uint8_t i=0; i<get_count(); i++) {
-        all_configured = all_configured && (!use_for_yaw(i) || configured(i));
+    for(uint8_t i=0; i<COMPASS_MAX_INSTANCES; i++) {
+        if (instance_available(i)) {
+            all_configured = all_configured && (!use_for_yaw(i) || configured(i));
```

Comment:
this is where we need to think about what we want to do for an arming check if a compass is removed. eg. if 2nd compass is removed out of 3. currently we'd fail arming checks. If I read this right then this change would make that case pass arming checks